migrations/fix_reverse_proxy_field/main.go

```diff
@@ -0,0 +1,124 @@
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type oldReservationData struct {
+	Description             string                          `bson:"description" json:"description"`
+	Currencies              []string                        `bson:"currencies" json:"currencies"`
+	SigningRequestProvision workloads.SigningRequest        `bson:"signing_request_provision" json:"signing_request_provision"`
+	SigningRequestDelete    workloads.SigningRequest        `bson:"signing_request_delete" json:"signing_request_delete"`
+	Containers              []workloads.Container           `bson:"containers" json:"containers"`
+	Volumes                 []workloads.Volume              `bson:"volumes" json:"volumes"`
+	Zdbs                    []workloads.ZDB                 `bson:"zdbs" json:"zdbs"`
+	Networks                []workloads.Network             `bson:"networks" json:"networks"`
+	Kubernetes              []workloads.K8S                 `bson:"kubernetes" json:"kubernetes"`
+	Proxies                 []workloads.GatewayProxy        `bson:"proxies" json:"proxies"`
+	ReserveProxy            []workloads.GatewayReverseProxy `bson:"reverse_proxies" json:"reserve_proxies"`
+	Subdomains              []workloads.GatewaySubdomain    `bson:"subdomains" json:"subdomains"`
+	DomainDelegates         []workloads.GatewayDelegate     `bson:"domain_delegates" json:"domain_delegates"`
+	Gateway4To6s            []workloads.Gateway4To6         `bson:"gateway4to6" json:"gateway4to6"`
+	ExpirationProvisioning  schema.Date                     `bson:"expiration_provisioning" json:"expiration_provisioning"`
+	ExpirationReservation   schema.Date                     `bson:"expiration_reservation" json:"expiration_reservation"`
+}
+
+func connectDB(ctx context.Context, connectionURI string) (*mongo.Client, error) {
+	client, err := mongo.NewClient(options.Client().ApplyURI(connectionURI))
+	if err != nil {
+		return nil, err
+	}
+
+	if err := client.Connect(ctx); err != nil {
+		return nil, err
+	}
+
+	return client, nil
+}
+
+func main() {
+	app.Initialize()
+
+	var (
+		dbConf string
+		name   string
+	)
+
+	flag.StringVar(&dbConf, "mongo", "mongodb://localhost:27017", "connection string to mongo database")
+	flag.StringVar(&name, "name", "explorer", "database name")
+	flag.Parse()
+
+	ctx := context.TODO()
+
+	client, err := connectDB(ctx, dbConf)
+	if err != nil {
+		log.Fatal().Err(err).Msg("failed to connect to database")
+	}
+	defer client.Disconnect(ctx)
+
+	db := client.Database(name, nil)
+	col := db.Collection(types.ReservationCollection)
+
+	cur, err := col.Find(ctx, bson.D{})
+	if err != nil {
+		log.Fatal().Err(err).Send()
+	}
+	defer cur.Close(ctx)
+
+	for cur.Next(ctx) {
+		r := types.Reservation{}
+		if err := cur.Decode(&r); err != nil {
+			log.Error().Err(err).Send()
+			continue
+		}
+
+		var data oldReservationData
+
+		if err := json.Unmarshal([]byte(r.Json), &data); err != nil {
```

Comment:
You should not rely on the data from the r.JSON field. Just load the data from the cursor in a variable of type oldReservationData, load the same data into a variable of the new type, and just copy the ReverseProxy from the old type to the new one.

Reply:
@zaibon how do you know which document to update then? Since the reservationData has no id?

Reply:
Threre was no need to load the data at all. Mongo has a `$rename` function. I've udapte the migration script.